src/game.h

```diff
@@ -30,6 +30,7 @@ extern bool trigdist;
 extern bool use_tiles;
 extern bool fov_3d;
 extern bool tile_iso;
+extern bool proper_stairs;
```

Comment:
The problem is that some of those need to be acquired really, really often:
* `trigdist` is read thousands of times per turn
* `fov_3d` is read O(`creature_num^2`) times per turn

Now that I think about it, `proper_stairs` doesn't need to be cached: it's read at most once per stair, per pathfinding call.
I'll get rid of its cached `bool`.